standards/naming.md

```diff
@@ -118,56 +118,119 @@ collectedItem: []
+    "bookId": "5196ab21",                       // since the request for the resource was a "book", the attribute should just be "id"
     "name": "The Best Book",
     "author": {                    
-        "authorId":456,                         // the parent entity is an author, so it should be "id" for the attribute name.
+        "authorId": "3793213e",                 // the parent entity is an author, so it should be "id" for the attribute name.
         "authorName": "John Doe",               // the parent entity is an author, so it should be "name" for the attribute name.
         "publisherName": "My Publishing House"  // this name attribute MUST have a prefix for another entity, since its not the name of this entity "author"
-        "agentId": 2                            // this name is appropriate with Id following its associated entity name.
     }
 }
  
 // CORRECT EXAMPLE
-GET /v1/books/2345
+GET /v1/books/5196ab21
 RESPONSE
 {
-    "id": 2345,                            
+    "id": "5196ab21",                            
     "name": "The Best Book",
     "author": {                    
-        "id":456,                      
+        "id": "3793213e",                      
         "name": "John Doe",            
-        "publisherName": "My Publishing House",
-        "agentId": 2
+        "publisherName": "My Publishing House"
     }
 }
 ```
 
+## Identifiers
+
+- An entity with a unique identifier **MUST** be labeled as `id`.
+
+```
+// CORRECT
+{
+    "id": "b96cb3ead9a9",                            
+    "name": "The Best Book"
+}
+```
+
+- Any unique identifier of an entity **MUST** have Id after the entity name when referenced from another entity.
+
+```
+// CORRECT
+{
+    "id": "b96cb3ead9a9",
+    "name": "The Best Book",
+    "authorId": "3793213e"      // usage of "authorId" references this is the id for a different entity.
+}
+```
+
+- Unique identifiers **SHOULD** be string values and considered opaque as described in [serialization of numbers](serialization.md#number).
+
+```
+// INCORRECT
+{
+    "id": 2
+}
+
+// CORRECT
+{
+    "id": "b96cb3ead9a9"
+}
+```
+
+## Domain References
+
+- Unique references to other significant entities within your API Domain, but in a different root resource **SHOULD** use [URN (Uniform Resource Name)](https://en.wikipedia.org/wiki/Uniform_Resource_Name)-like references where appropriate.  
+    - URN-like refers to using the convention of specifying agreed upon namespaces, entities followed by the object identifier, but without the `urn` prefix or formal registration of the namespace ID with IANA according to [RFC8141](https://datatracker.ietf.org/doc/rfc8141/). This provides interoperability and durability benefits within your API and endpoint ecosystem.
+    - URN-like values **MUST** the form of standard `URNs` without the `urn:` prefix to avoid confusion of official registration. A simplified example would be: `{namespace}:{entity}:{id}` (e.g. `spsc:book:b96cb3ead9a9`).
```

Comment:
Are we... going to talk about `id`'s structure?  Or are we holding off on that.  Or addressing it in that other issue.

Reply:
Other than the total URN-like value fitting in under 255 characters, can you help me understand some of the ID restrictions that would be placed here?

Reply:
The idea that we should force ids to look similar to Identity's id's (the uuid encoded as a string).  Something that's had conversation in the past.  I think we left off with needing a good two-way hash function when the underlying id wasn't a uuid.

Reply:
Ya, I was considering whether that belonged here. The idea would be that the object id is NOT a number (i.e. spsc:item:3). This matching up to the associated "id" would automatically imply that given the recommendations already around ids.

Given the content we have on IDs already, I'm thinking we should just leave that out here.

Reply:
Cool.  That sounds good to me.  Anyway to link up to those recommendations?  Or maybe that's already in there...